### PR TITLE
chore: Use get-tested for GH actions

### PR DESCRIPTION
**Pull request by CristhianMotoche**
# Modified
- Updated directory and optparse-applicative boundaries
- Include tested-with
- Modify GH Actions

solves #51

### CODE REVIEW

**Review by DavidMazarro — APPROVED** [review]
LGTM! Left a small suggestion. Not sure why the CI didn't trigger after your last commit though.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -14,22 +14,45 @@ concurrency:
...
+    name: "Generate matrix from cabal"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Extract the tested GHC versions
+        id: set-matrix
+        uses: kleidukos/get-tested@v0.1.7.0
+        with:
+          cabal-file: get-tested.cabal
```
Comment (DavidMazarro) [inline .github/workflows/build-and-test.yml:29]: I missed this in the previous review, my bad! The `cabal-file` needs to point to the project's Cabal file.
```suggestion
          cabal-file: hspec-golden.cabal
```

**Review by DavidMazarro — CHANGES_REQUESTED** [review]
Requesting changes (left a suggestion) because otherwise the CI will not work.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (CristhianMotoche) [reply .github/workflows/build-and-test.yml]: Thanks for catching that, David!

**Comment by CristhianMotoche** [comment]
Hey @DavidMazarro I think this is ready for a second review!

**Review by DavidMazarro — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**